python/rmm/tests/test_rmm.py

```diff
@@ -942,3 +943,73 @@ def test_rmm_device_buffer_copy(cuda_ary, make_copy):
     result = db_copy.copy_to_host()
 
     np.testing.assert_equal(expected, result)
+
+
+@pytest.mark.parametrize("level", rmm.logging_level)
+def test_valid_logging_level(level):
+    warnings.filterwarnings(
+        "ignore", message="RMM will not log warning level TRACE."
+    )
+    warnings.filterwarnings(
+        "ignore", message="RMM will not log warning level DEBUG."
+    )
+    rmm.set_logging_level(level)
+    assert rmm.get_logging_level() == level
+    # Also test setting using any value or name in the enum
+    rmm.set_logging_level(level.value)
```

Comment:
<!-- Polish comments are like a suggestion, where there is nothing necessarily wrong with the relevant content, there's just some ways to immediately improve the quality.  -->
**polish:** ‏It _could_ be the case that the `set_logging_level` on line 956 succeeds, and then the one on 959 fails silently, but we would not notice because we're still expecting the same level.

One way to handle this would be to parameterize the test over all the possible things we want to test. If we do this, I would then recommend making a reusable fixture (so we can use the same thing in `test_valid_logging_flush_level`):

```python
@pytest.fixture(params=["enum", "integer", "string", "case-insensitive"])
def level_type(request):
    return request.param


@pytest.fixture(params=rmm.logging_level)
def level(request, level_type):
    # return value is (thing_well_test, expected)
    level = request.param
    if level_type == "enum":
        return (level, level)
    elif level_type == "integer":
        return (level.value, level)
    elif level_type == "string":
        return (level.name, level)
    elif level_type == "case-insensitive":
        return (level.name.title(), level)


def test_valid_logging_level(level):
    level, expect = level
    rmm.set_logging_level(level)
    assert rmm.get_logging_level() == expect
    # reset to default
    rmm.set_logging_level("INFO")


def test_valid_logging_flush_level(level):
    level, expect = level
    rmm.set_flush_level(level)
    assert rmm.get_flush_level() == expect
    # reset to default
    rmm.set_logging_level("INFO")
```

Reply:
I've drastically simplified this because we no longer ducktype, and only support passing the enum instances themselves.

Reply:
Thanks, looks good